src/main/sensors/rpm_filter.c

```diff
@@ -121,6 +122,9 @@ void rpmFilterInit(const rpmFilterConfig_t *config)
     numberFilters = getMotorCount() * (filters[0].harmonics + filters[1].harmonics);
     const float filtersPerLoopIteration = numberFilters / loopIterationsPerUpdate;
     filterUpdatesPerIteration = rintf(filtersPerLoopIteration + 0.49f);
+    if (gyroFilter || dtermFilter) {
+        schedulerOptimizeRate(true);
```

Comment:
This is the wrong place to change the configuration - for a user who disables the scheduler rate optimisation, it will be confusing if his setting is silently ignored. This also makes debugging / tuning pretty much impossible, as the result of a `diff` is not an actual representation of the configuration that is used.

If settings are interdependent, then this should be addressed in `validateAndFixConfig()`, to make it permanent and transparent to the user.

Reply:
agreed

Reply:
Ok. If I do it in validateAndFixConfig though it means that enabling and then disabling the rpm filter will permanently enable the schedulerOptimizeRate feature. That's then ok, right?

Reply:
The way that this is done in other cases is that options are disabled if the prerequisites aren't met, so enabling the rpm filter without `schedulerOptimizeRate` should disable the rpm filter.

Reply:
If we do that it means that on set dshot_bidir=true the rpm_filter would automatically deactivate since schedulerOptimizeRate is off by default. The rpm_filter is deactivated by setting the harmonics setting to 0, which means that the harmonics defaults get reset. A pretty big impact in usability.

Would you be ok with eliminating the schedulerOptimizeRate setting altogether and just always enable it together with the rpmFilter?

Reply:
If we ever hope to make `schedulerOptimizeRate` the default then we should make it possible to enable it standalone, so that users that cannot run rpm filter can test it.

I don't think setting harmonics to 0 will be a big impact - the default will always be visible when queried in CLI.

Reply:
I think I found an acceptable solution: disabling dshot_bidir if schedulerOptimizeRate isn't set.